### ISSUE TITLE
No In App updates onResume

### ISSUE THREAD

**Issue by jeremyeaton89** [microsoft/appcenter-sdk-android#1321]
### **Description**

I integrated the SDK (pretty simple setup) and included the Distribute module. I do get in-app updates, but only after killing the initial build running app center sdk on the device and relaunching. 

The iOS SDK _does_ do this properly, as did our previous Android Hockeyapp SDK powered app. 

### **Repro Steps**

Please list the steps used to reproduce your issue.

1. push up a test build to app center
2. install that build on device; launch it to setup in-app updates
3. push up another build with incremented bundle and version
4. expect in app update prompt, but none. background the app, then foreground it — still no in app update
5. kill app; launch again. Now we see the in-app update prompt. 

### **Details**

1. Which SDK version are you using?
    2.5.0
2. Which OS version did you experience the issue on?
    -Android 9
3. What device version did you see this error on?  Were you using an emulator or a physical device?
    Samsung Galaxy S8
4. What third party libraries are you using?
    This is a React Native app if that makes any difference. The RN SDK didn't have support Distribute/in-app-update support, so went native SDK
5. I have done just that but would like to pass this along privately as there may be sensitive info there.

**Comment by fabiendem** [comment]
I had a similar issue because I was installing the app via `adb` rather than from the AppCenter website.
Check this thread (it was with an older SDK though): https://github.com/microsoft/appcenter/issues/1205#issuecomment-546380578

Maybe you're facing the same issue?

**Comment by guperrot** [comment]
It's actually by design where we didn't want to make an API call for every screen change so right now it reset the state only at onCreate of a launcher activity.

**Comment by guperrot** [comment]
An improvement could be to also monitor onResume but with some limitations around it to avoid too many calls.

**Comment by jeremyeaton89** [comment]
@guperrot i see. I wonder if it's because I'm running React Native that i've had to kill the whole app to see a prompt. 

Is Distribute functionality happening any time soon for React Native SDK?

**Comment by jeremyeaton89** [comment]
@fabiendem thanks for the comment. I've been uploading to app center and downloading app from email.

@guperrot i've also had difficulty testing the in-app update functionality when going from hockeyapp SDK 5.0.0 to app center. i will delete this comment and post new issue if need be, but it _is_ expected that pushing a build to appcenter will trigger in app update for that distribution group even if user's app is running Hockeyapp? I haven't tried making the distribution channel i'm testing on public because it seems risky, but might it need to be public?

**Comment by fabiendem** [comment]
>  I wonder if it's because I'm running React Native that i've had to kill the whole app to see a prompt.

Is Distribute functionality happening any time soon for React Native SDK?
I raised a similar issue before
https://github.com/microsoft/appcenter/issues/1137
(https://github.com/microsoft/appcenter-sdk-android/issues/1313)

> @fabiendem thanks for the comment. I've been uploading to app center and downloading app from email.

Ok, from what I remember, I can at least retrigger the check of the update by a tap on the icon of the app (rather than a resume from the recent apps), but that may be specific to the lifecycle of my app.
Am on React Native as well and did a native integration like you did.

**Comment by guperrot** [comment]
A known caveat is just pressing HOME then resuming the app again, it does not count as a restart.
Only BACK from main activity then restarting activity will. If that does not work even with BACK it might be a problem of life cycle not calling onDestroy then onCreate again on the main activity.

The feature request can be kept open to improve the behavior.

For React Native Distribute there is a feature request you can vote for: https://github.com/microsoft/appcenter-sdk-react-native/issues/225

For manual update checks there is another feature request: https://github.com/microsoft/appcenter/issues/1137

**Comment by jeremyeaton89** [comment]
@guperrot will this improvement/feature gap be implemented prior to the shutdown of Hockeyapp?

**Comment by ahdbilal** [comment]
@jeremyeaton89 Currently, it is not in short term plan but I will escalate this to the right team to have it considered before HockeyApp final shutdown. I will post here once I get an update. Thanks

**Comment by Corical** [comment]
Well maybe it is acceptable for a _normal_ application, but what about a security application that is locked down (pinned / task locked) and only has one Activity. The application is supposed to run 24/7/365. So what would you propose in this situation ? Again, this was supported by HockeyApp.

I already had to intercept your distribution process at `onReleaseAvailable` to download the APK myself and then launch the installer myself because of this issue -> https://github.com/microsoft/appcenter-sdk-dotnet/issues/787

Feels like as soon as I solve something you missed I hit another brick wall. This is really a deal breaker guys and gals.

**Comment by jeremyeaton89** [comment]
@ahdbilal do we know if this will be addressed before imminent shutdown Feb 10th?

**Comment by ahdbilal** [comment]
I don't see this being released before the shutdown. I will keep on pushing it internally but I can't share a timeline as of now. Once it moves in active stage, I will share an update here.
